src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+    EVENT_ORC_PUP_REMOVE_PROTECTION             = 2,
+    EVENT_ORC_PUP_TALK                          = 3,
+    EVENT_ORC_PUP_DELIVER_CRATES                = 4,
+
+    TALK_ORC_PUP_SUMMONED                       = 0,
+    TALK_ORC_PUP_DELIVER_CRATES                 = 6,
+    TALK_ORC_PUP_WORN_OFF                       = 7
+};
+
+// 44914 - Orc Sea Pup
+struct npc_silverpine_orc_sea_pup : public VehicleAI
+{
+    npc_silverpine_orc_sea_pup(Creature* creature) : VehicleAI(creature), _isJustSummoned(true) { }
+
+    void JustAppeared() override
+    {
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+        {
+            _playerGUID = player->GetGUID();
+
+            me->RemoveNpcFlag(UNIT_NPC_FLAG_SPELLCLICK);
+
+            me->GetMotionMaster()->MoveFollow(player, 2.0f, float(M_PI / 2.0f));
+
+            if (player->GetQuestStatus(QUEST_STEEL_THUNDER) == QUEST_STATUS_INCOMPLETE)
+            {
+                _events.ScheduleEvent(EVENT_ORC_PUP_JUST_SUMMONED, 1s);
+                _events.ScheduleEvent(EVENT_ORC_PUP_REMOVE_PROTECTION, 1s + 500ms);
+                _events.ScheduleEvent(EVENT_ORC_PUP_TALK, 5s);
+
+                int c = player->GetReqKillOrCastCurrentCount(QUEST_STEEL_THUNDER, NPC_ORC_CRATE);
+
+                for (int i = 0; i < c; i++)
+                    DoCastSelf(SPELL_SUMMON_ORC_CRATE);
+            }
+        }
+    }
+
+    void PassengerBoarded(Unit* passenger, int8 seatId, bool apply) override
+    {
+        if (passenger->GetEntry() == NPC_ORC_CRATE)
+        {
+            if (apply)
+            {
+                if (_isJustSummoned)
+                    return;
+
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    Talk(seatId + 1, player);
+            }
+            else
+                passenger->ToCreature()->DespawnOrUnsummon(3s);
+        }
+    }
+
+    void SpellHit(WorldObject* /*caster*/, SpellInfo const* spellInfo) override
+    {
+        switch (spellInfo->Id)
+        {
+            case SPELL_SEA_PUP_TRIGGER:
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    Talk(TALK_ORC_PUP_DELIVER_CRATES, player);
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    player->CastSpell(nullptr, SPELL_DESPAWN_ALL_SUMMONS, true);
+                _events.CancelEvent(EVENT_ORC_PUP_TALK);
+                break;
+
+            case SPELL_DESPAWN_ALL_SUMMONS:
+                _events.ScheduleEvent(EVENT_ORC_PUP_DELIVER_CRATES, 1s + 250ms);
+                break;
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+        {
+            if (!player->HasAura(SPELL_SUMMON_ORC_SEA_PUP))
```

Comment:
can we shovel that into a generic spellscript with `OnRemove` somehow? i kinda dont like iterating all player auras just for this every tick

Reply:
Neither do I but there's missing core support for many SPELL_EFFECT_SUMMON effects which are bound to a SPELL_EFFECT_APPLY_AURA as SPELL_AURA_DUMMY. 

I don't know if there's any documentation anywhere on TC about this, but as of today, retail has dynamic handling for these in which:

- if the player somehow loses the aura, either by dying or moving out of an allowed aura (e.g. spell_area), the creature should be unsummoned.
- if the creature dies, there's a serverside timer at which the creature is resummoned (not sure if by using the same auraGUID or using a new one by removing the aura and forcing the caster to cast it again).


On TC, it works this way:

- if the summon creature happens to die or despawn due to distance while the aura persists, it won't spawn ever again until the player manually uses .cast to summon it.
- if the player happens to log off and back in into the world and meets the requirements to cast the spell through spell_area, but the player also met the requirements before logging off, the creature related to the SPELL_EFFECT_SUMMON won't be summoned. This is probably due to the fact that the aura was still on the player (though it must be noted that the NPC will be spawned if the player has the aura and they cast the spell through .cast, but not through spell_area).
- if the player doesn't meet the requirements anymore, e.g. going out of the area related to the spell on spell_area or the zone id or area id bound to the spell, the NPC won't unsummon.

Reply:
offl found something cool in ur other pr, maybe its the case here aswell: https://github.com/TrinityCore/TrinityCore/pull/28334#discussion_r990509726

Reply:
![image](https://user-images.githubusercontent.com/30064812/194702666-78031616-b916-458c-adc9-02903f4dfaa5.png)
The problem is, even with these flags, the aura still persists and the creature won't be respawned.

Reply:
Feedback applied.